resourcemanager/resource_manager.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright (c) 2023, Izuma Networks
+
+SPDX-License-Identifier: Apache-2.0
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package resourcemanager provides the interface to the gateway resource manager API
+package resourcemanager
+
+import (
+    b64 "encoding/base64"
+    "encoding/json"
+    "errors"
+    "io/ioutil"
+    "os"
+    "github.com/PelionIoT/edge-resource-manager/resourcemanagerconfig"
+    "github.com/op/go-logging"
+)
+
+var log = logging.MustGetLogger("EdgeResourceManager")
+
+type gwResourceManagerRegisterArgs struct {
+    Name string `json:"name"`
+}
+
+type addResouceArgs struct {
+    Lwm2mObjects []lwm2mObject `json:"objects"`
+}
+
+type lwm2mObject struct {
+    ObjectID        int              `json:"objectId"`
+    ObjectInstances []objectInstance `json:"objectInstances"`
+}
+
+type objectInstance struct {
+    ObjectInstanceID int        `json:"objectInstanceId"`
+    Resources        []resource `json:"resources"`
+}
+
+type resource struct {
+    ResourceID int    `json:"resourceId"`
+    Operations int    `json:"operations"`
+    Type       string `json:"type"`
+    Value      string `json:"value"`
+}
+
+var resourceManagementClient *Client = nil            // Edge Resource Manager Client
+var gcdConfig *resourcemanagerconfig.ResourceManagerConfig // Edge Capability Discovery config
+
+// Run provides the main execution of the program. It adds resources th the gateway as detailed in the config file and handles updates.
+func Run(config *resourcemanagerconfig.ResourceManagerConfig) {
+    log.Info("resourceManager.Run")
+
+    if config == nil {
+        log.Warning("ResourceManager: No config Provided\n")
+        return
+    }
+    if config.EdgeCoreSocketPath == "" {
+        log.Warning("ResourceManager: edge_core_socketpath not provided in config file. Using default socketpath '/tmp/edge.sock'\n")
+        config.EdgeCoreSocketPath = "/tmp/edge.sock"
+    }
+    if config.EdgeResources == nil {
+        log.Errorf("ResourceManager: No edge resources provided\n")
+        return
+    }
+    if config.ConfigObjectID <= 0 {
+        log.Errorf("ResourceManager: lwm2m objectid not provided\n")
+        return
+    }
+    var err error
+    gcdConfig = config
+    log.Infof("ResourceManager: connecting to edge-core\n")
```

Comment:
Do you really need to add the  `\n`'s there? Could the linechanges be behind some API instead?